e2e_samples/fabric_dataops_sample/docs/adr_naming_convention.md

```diff
@@ -0,0 +1,79 @@
+
+## Proposal details
+
+### GitHub files naming convention
+
+| Item Type | Convention | Additional Details | Example |
+| --- | --- | --- | --- |
+| Folders | snake_case | Use lowercase and "_" | `fabric_dataops_sample`, `dataops_pipeline` |
+| Markdown files | snake-case | Use lowercase and "_" | "README.md", `adr_naming_convention.md` |
+| Python scripts | snake_case | Use lowercase and "_" | `setup_fabric_environment.py` |
+| Shell scripts | snake_case | Use lowercase and "_" | `deploy_infrastructure.sh` |
+| PowerShell scripts | PascalCase | Use "-" and "verb-noun" structure | `Set-DeploymentPipelines.ps1` |
+| Images (diagrams, screenshots etc.) | kebab-case | Use lowercase and "-" | `graph-api-permission.png`, `fabric-architecture.drawio`|
+| YAML files | kebab-case | Use lowercase and "-" | `deploy-infra.yml` |
+| JSON files | kebab-case | Use lowercase and "-" | `fabric-config.json` |
+| Configuration files | kebab-case | Use lowercase and "-" | `dataops-config.cfg` |
+| Wheel files | [pep-427](https://peps.python.org/pep-0427/) | {distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl | `azure_monitor_opentelemetry-1.6.1-py3-none-any.whl` |
+
+### Variables/Functions naming convention
+
+| Variable Type | Convention | Additional Details | Example |
+| --- | --- | --- | --- |
+| Environment variables | UPPER_SNAKE_CASE | Use uppercase and "_" | `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET` |
+| Shell script variables to read Terraform output | `tf_`snake_case | Prefix with "tf_" | `tf_storage_account_name` |
+| Shell/Python script variables (in general) | snake_case | Use lowercase and "_" | `storage_account_name` |
+| PowerShell script variables | `$`camelCase |  | `$storageAccountName` |
+| Shell/Python Functions | snake_case | Use lowercase and "_" | `get_storage_account_name` |
+| PowerShell Functions | PascalCase | Use "Verb-Noun" structure | `Get-StorageAccountName` |
+| Terraform variables | snake_case | Use lowercase and "-" | `environment_name` |
+
+### Azure/Fabric resource naming convention
+
+| Resource/Item Name | Convention | Default Name |
+| --- | --- | --- |
+| Resource Group | `rg-*` | N/A - passed |
+| ADLS Gen2 Account | `st*` | `st<base-name>` |
+| Storage container name | N/A | "main" |
+| Azure Key Vault | `kv-*` | `kv-<base-name>` |
+| Entra Security Group | `sg-*` | N/A - passed |
+| Log Analytics Workspace | `la-*` | `law-<base-name>` |
+| Application Insights | `appi-*` | `appi-<base-name>` |
+| Fabric Capacity | `cap*` | `cap<base-name-trimmed>` |
+| Fabric Workspace | `ws-*` | `ws-<base-name>` |
+| Fabric Lakehouse | `lh_*` | `lh_<base-name>` |
+| Fabric Environment | `env-*` | `env-<base-name>` |
+| Fabric Notebooks | `nb-*` | "nb-010-setup.ipynb", "nb-020-standardize.ipynb", "nb-030-transform.ipynb" |
```

Comment:
I still don't understand why are we using 010/020/030 - can you share the logic?

Reply:
@ydaponte , "0x0" gives the flexibility of adding additional functionality between different processes. For example: if we decide to write another notebook just for the validations after setup but before standardize, we can plug it as "nb-011-validation.ipynb" notebook. This retains ordering and adds flexibility.

But I also get your point about being consistent. Happy to rename it to "0x_" as well. Let's discuss in a meeting.